### PR TITLE
Feature/2216 grafana endpoint to see how many ecr repos are out of sync

### PR DESCRIPTION
**Pull request by paulseghers**
closes dfds/cloudplatform/issues/2216

# Additional Review Notes
Because of low knowledge I ended up going down the "wrong" path for 70% of this PR, however I'm still leaving the code for this.
Essentially we are exposing the metrics in two ways:
- a `Prometheus.Metrics.Gauge` which comes with its own prometheus .net functions to update it
- an in-memory global variable which gets exposed on a conventional http endpoint 

Only the first one is actually needed, but I've made a background service which updates both ¯\_(ツ)_/¯

### CODE REVIEW

File: src/SelfService/Infrastructure/Api/Metrics/MetricsController.cs [hunk 1]
```diff
@@ -64,6 +69,52 @@ public async Task<IActionResult> GetMyCapabilitiesCosts()
...
+        }
+
+        return NotFound(
+            new ProblemDetails()
+            {
+                Title = "Capability Costs not found",
+                Detail = $"No Cost data found for any capability",
+            }
+        );
+    }
```
Comment (kralle333) [inline src/SelfService/Infrastructure/Api/Metrics/MetricsController.cs:116]: copy paste mistakes

File: src/SelfService/Infrastructure/BackgroundJobs/UpdateOutOfSyncECRRepos.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        return Task.Run(
+            async () =>
+            {
+                while (!stoppingToken.IsCancellationRequested)
+                {
+                    await DoWork(stoppingToken);
+                    await Task.Delay(TimeSpan.FromMinutes(1), stoppingToken); //maybe each minute is a lot?
```
Comment (kralle333) [inline src/SelfService/Infrastructure/BackgroundJobs/UpdateOutOfSyncECRRepos.cs:25]: I would make it lazy instead. Just update on need if its older than some amount of time (12 hours?)

**Review by kralle333 — CHANGES_REQUESTED** [review]
mostly good, but I have some comments

File: src/SelfService/Infrastructure/Api/Metrics/MetricsController.cs (hunk 1 above)
Comment (paulseghers) [reply src/SelfService/Infrastructure/Api/Metrics/MetricsController.cs:116]: thx

**Review by kralle333 — APPROVED** [review]
good

**Pull request merged**